platforms/jvm/plugins-java-application/src/main/java/org/gradle/api/plugins/JavaApplicationPlugin.java

```diff
@@ -59,7 +59,8 @@
  *
  * @see <a href="https://docs.gradle.org/current/userguide/application_plugin.html">Application plugin reference</a>
  */
-public abstract class ApplicationPlugin implements Plugin<Project> {
+@SuppressWarnings("deprecation")
+public abstract class JavaApplicationPlugin extends ApplicationPlugin implements Plugin<Project> {
```

Comment:
❌ This doesn't quite work for people applying `ApplicationPlugin` by class instead of id.

I think you have to actually leave everything in ApplicationPlugin and extend it here with the new name.